AreaShop/src/main/java/me/wiefferink/areashop/ASLanguageManager.java

```diff
@@ -0,0 +1,164 @@
+package me.wiefferink.areashop;
+
+import com.google.common.base.Charsets;
+import me.wiefferink.interactivemessenger.Log;
+import me.wiefferink.interactivemessenger.processing.Message;
+import me.wiefferink.interactivemessenger.source.LanguageManager;
+import me.wiefferink.interactivemessenger.translation.Transifex;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.plugin.java.JavaPlugin;
+
+import java.io.*;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class ASLanguageManager extends LanguageManager
+{
+    private JavaPlugin plugin;
+    private String jarLanguagePath;
+    private File languageFolder;
+    private String wgPrefix;
+    private List<String> chatPrefix;
+    private Map<String, List<String>> currentLanguage, defaultLanguage;
+
+    public ASLanguageManager(JavaPlugin plugin, String jarLanguagePath, String currentLanguageName, String defaultLanguageName, List<String> chatPrefix, String wgPrefix)
+    {
+        super(plugin, jarLanguagePath, currentLanguageName, defaultLanguageName, chatPrefix);
+        this.plugin = plugin;
+        this.jarLanguagePath = jarLanguagePath;
+        this.chatPrefix = chatPrefix;
+        this.languageFolder = new File(plugin.getDataFolder() + File.separator + jarLanguagePath);
+
+        Message.init(this, plugin.getLogger());
+        saveDefaults();
+        currentLanguage = loadLanguage(currentLanguageName);
+        if(defaultLanguageName.equals(currentLanguageName)) {
+            defaultLanguage = currentLanguage;
+        } else {
+            defaultLanguage = loadLanguage(defaultLanguageName);
+        }
+        this.wgPrefix = wgPrefix;
+    }
+
+    @Override
+    public List<String> getMessage(String key) {
+        List<String> message;
+        if(key.equalsIgnoreCase(Message.CHATLANGUAGEVARIABLE)) {
+            message = chatPrefix;
+        } else if(key.equalsIgnoreCase("wgPrefix"))
+        {
+            message = new ArrayList<>();
+            message.add(wgPrefix);
+        } else if(currentLanguage.containsKey(key)) {
+            message = currentLanguage.get(key);
+        } else {
+            message = defaultLanguage.get(key);
+        }
+        if(message == null) {
+            Log.warn("Did not find message '" + key + "' in the current or default language");
+            return new ArrayList<>();
+        }
+        return new ArrayList<>(message);
+    }
+
+    private void saveDefaults() {
```

Comment:
I assume you've copied this straight from the LanguageManage super method, have you changed anything here? If you haven't you might as well refactor it to use the NIO since we're on Java 17 now.

Reply:
It was just copied from LanguageManager but it was private so I couldn't just call super.saveDefaults() and I didn't want to use Reflection.

Reply:
I see. Well, nonetheless, since you're adding it in, I'd like for this method to be refactored such that it uses NIO. It seems to be a trivial refactor anyway that saves a lot of headache in terms of readability